R/list_animal_ids_api.R

```diff
@@ -0,0 +1,27 @@
+#' List all available animal ids
+#'
+#' @param credentials Login credentials to the ETN database, as created by
+#'   `get_credentials()`
+#'
+#' @return A vector of all unique `id_pk` present in `common.animal_release`.
+#'
+#' @export
+list_animal_ids_api <- function(credentials = get_credentials()) {
+  api_domain <- "https://opencpu.lifewatch.be"
+  function_path <- "library/etn/R/list_animal_ids"
+
+  # POST the function request to the api_domain, request json as direct return
+  response <-
+    httr::POST(stringr::str_glue(api_domain, function_path, "json", .sep = "/"),
```

Comment:
Please use `glue::glue()` over `stringr::str_glue()`. We use glue directly in our packages, as it is very lightweight (no dependencies), while stringr is a more of a conscious dependency.

Reply:
Note that here we could also use `paste()`. I typically use `paste()` for simple things like this and`glue()` when I want to insert variables.